examples/widgets/annotated_cursor.py

```diff
@@ -0,0 +1,345 @@
+            *None* is the fallback value.
+        """
+
+        # Get plot line data
+        xdata = self.line.get_xdata()
+        ydata = self.line.get_ydata()
+
+        # The dataaxis attribute decides, in which axis we look up which cursor
+        # coordinate.
+        if self.dataaxis == 'x':
+            pos = xpos
+            data = xdata
+            lim = self.ax.get_xlim()
+        elif self.dataaxis == 'y':
+            pos = ypos
+            data = ydata
+            lim = self.ax.get_ylim()
+        else:
+            raise ValueError(
+                "The data axis specifier {} should be 'x' or 'y'"
+                .format(self.dataaxis))
+
+        # If position is valid
+        if pos is not None:
+            # And in valid plot data range
+            if pos >= lim[0] and pos <= lim[-1]:
+                # Convert given positon to numpy array,
+                # so numpy function can be used.
+                findme = np.array([pos])
+                # Find closest x value in sorted x vector.
+                # This is the code line,
+                # which requires the plotted data to be sorted.
+                index = np.searchsorted(data, findme)
+                # Return none, if this index is out of range.
+                if (index < 0) or (index >= len(data)):
+                    return None
+                # Return plot point as tuple.
+                return (xdata[index][0], ydata[index][0])
+
+        # Return none if there is no good related point for this x position.
+        return None
+
+    def clear(self, event):
+        """Overwirtten clear callback for cursor. Called right before \
+        displaying the figure."""
+
+        # The base class saves the clean background for blitting.
+        # Text and cursor are invisible,
+        # until the first mouse move event occurs.
+        super().clear(event)
+        if self.ignore(event):
+            return
+        self.text.set_visible(False)
+
+    def _update(self):
+        """
+        Overwritten method for eather blitting or drawing the widget canvas.
+
+        Passes call to base class if blitting is activated, only.
+        In other cases, one draw_idle call is enough, which is placed
+        explicitly in this class (see *onmove()*).
+        In that case, `~matplotlib.widgets.Cursor` is not supposed to draw
+        something using this method.
+        """
+
+        if self.useblit:
+            super()._update()
+
+
+fig, ax = plt.subplots(figsize=(8, 6))
+
+x = np.linspace(-5, 5, 1000)
+y = x**2
+
+line, = ax.plot(x, y)
+ax.set_xlim(-5, 5)
+ax.set_ylim(0, 25)
+
+# A minimum call
+# Set useblit=True on most backends for enhanced performance
+# and pass the ax parameter to the Cursor base class.
+# cursor = AnnotatedCursor(line=lin[0], ax=ax, useblit=True)
+
+# A more advanced call. Properties for text and lines are passed.
+# See the color if you are confused which parameter is passed where.
```

Comment:
What do you mean "See the color"?

Reply:
Imporved the explanation in b1c0b2498351827394c1229c18e7f28e0bcc5ea4